EIPS/eip-2135.md

```diff
@@ -1,80 +1,103 @@
+
+  /// @notice The interface to check whether an asset is consumable.
+  /// @param _consumer the address of consumer of this token. It doesn't have
+  ///                  to be the EOA or contract Account that initiates the TX.
+  /// @param _assetId  the NFT asset being consumed
+  function isConsumableBy(address _consumer, uint256 _assetId) external view returns (bool _consumable);
+
+  /// @notice The event emitted when there is a successful consumption.
+  /// @param _consumer the address of consumer of this token. It doesn't have
+  ///                  to be the EOA or contract Account that initiates the TX.
+  /// @param _assetId  the NFT asset being consumed
+  /// @param _data     extra data passed in for consume for extra message
+  ///                  or future extension.
+  event OnConsumption(address indexed _consumer, uint256 indexed _assetId, bytes[] _data);
 }
 ```
 
+2. If the compliant contract is an [EIP-721](./eip-721.md) or [EIP-1155](./eip-1155.md), in addition to `OnConsumption`, it **MUST** also emit related `Transfer` / `TransferSingle` event as if a token has been transfer from the current holder to `0x00` address
+if the call to `consume` method succeeds.
+
+3. The compliant contract **MUST** respond `true` for
+[EIP-165](./eip-165.md) identifier `interfaceID=0x0c44c20d` when queried
+`supportsInterface(bytes4 interfaceID)`
+
 ## Rationale
 
-The function `consume` performs the consume action. Being an interface standard, 
+1. The function `consume` performs the consume action. Being an interface standard,
 this EIP does not impose any assumption of
 
- - who has the power to perform such activity. 
- - under what condition such consumption can occur.
- 
-It does, however, assume the asset can be identified in a `uint256` assetId as in th parameter. A design convention and compatibility consideration is put in place to follow the ERC-721
+- who has the power to perform such activity.
+- under what condition such consumption can occur.
 
-The event notifies subscribers whoever are interested to learn an asset is being consumed. The boolean function of `isConsumable` can be used to check whether an asset is still consumable. 
+It does, however, assume the asset can be identified in a `uint256` assetId as in th parameter. A design convention and compatibility consideration is put in place to follow the EIP-721
 
-To keep it simple, this standard *intentionally* contains no functions or events related to creation of a consumable asset. This is because the creation of a consumable asset will need to make assumption of the nature of an actual use-case. If we see some common use-case of creation, we can have another follow up standard.
+2. The event notifies subscribers whoever are interested to learn an asset is being consumed. The boolean function of `isConsumable` can be used to check whether an asset is still consumable.
```

Comment:
If the asset is transferred to the zero address, is it ever consumable again?

Reply:
No, it will not be consumable again. You made a good point. Let me make it more clear.